scipy/optimize/tests/test_minpack.py

```diff
@@ -602,8 +602,9 @@ def _check_nan_policy(f, xdata_with_nan, xdata_without_nan,
         assert_allclose(result_with_nan, result_without_nan)
 
         # not valid policy test
-        error_msg = ("nan_policy must be one of "
-                     "{'None', 'raise', 'omit'}")
+        # check for argument names in any order
+        error_msg = (r"nan_policy must be one of \{(?:'raise'|'omit'|None)"
+                     r"(?:, ?(?:'raise'|'omit'|None))*\}")
```

Comment:
We could simplify this to just:
```suggestion
        error_msg = "nan_policy must be one of..."
```
The only reason I didn't is that `curve_fit` has a different set of `nan_policy` options from most functions, so it seemed reasonable to check them specifically. 
I don't often use Chat GPT, but it sure comes in handy for regular expressions...